packages/hawtio/src/plugins/jmx/JmxTreeView.tsx

```diff
@@ -6,23 +6,54 @@ import { MBeanTreeContext } from './context'
 import './JmxTreeView.css'
 
 export const JmxTreeView: React.FunctionComponent = () => {
-  const { tree, setNode } = useContext(MBeanTreeContext)
-  const [expanded, setExpanded] = useState(false)
+  const { tree, node, setNode } = useContext(MBeanTreeContext);
+  const [expanded, setExpanded] = useState(false);
+  const [filteredTree, updateTree] = useState(tree.getTree());
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const onSearch = (event: ChangeEvent<HTMLInputElement>) => {
-    // TODO
+    if(!event.target.value) updateTree(tree.getTree());
+    
+    const treeElements = lookupSearchInTree(event.target.value, tree.getTree());
+
+    if (treeElements.length == 0) updateTree(tree.getTree())
+    else updateTree(treeElements);
   }
 
   const onSelect = (event: React.MouseEvent<Element, MouseEvent>, item: TreeViewDataItem) => {
     setNode(item as MBeanNode)
   }
 
+  const lookupSearchInTree = (search: string, tree?: MBeanNode[]) : MBeanNode[] => {
+    if (tree?.length == 0) return [];
```

Comment:
It should also check `tree`'s existence.

Reply:
If tree is null or undefined, the ?. operator will return undefined instead of following.

It's equivalent to `if(tree && tree.length ==0)`